libraries/AP_Proximity/AP_Proximity_MAV.cpp

```diff
@@ -88,26 +88,36 @@ void AP_Proximity_MAV::handle_msg(mavlink_message_t *msg)
         mavlink_msg_obstacle_distance_decode(msg, &packet);
 
         // check increment (message's sector width)
-        uint8_t increment = packet.increment;
-        if (packet.increment == 0) {
+        float increment;
+        if (packet.increment_f > 0) {
+            // use increment float
+            increment = packet.increment_f;
+        } else if (packet.increment > 0) {
+            // use increment uint8_t
+            increment = packet.increment;
+        } else {
+            // invalid increment
             return;
         }
 
         const float MAX_DISTANCE = 9999.0f;
-        const uint8_t total_distances = MIN(360.0f / increment, 72);
+        uint8_t count = MAVLINK_MSG_OBSTACLE_DISTANCE_FIELD_DISTANCES_LEN;
+        if (packet.count > 0 &&  packet.count < MAVLINK_MSG_OBSTACLE_DISTANCE_FIELD_DISTANCES_LEN) {
+            count = packet.count;
+        }
+        const float total_distances = MIN(360.0f / increment, count); // usually 72
+
 
         // set distance min and max
         _distance_min = packet.min_distance / 100.0f;
         _distance_max = packet.max_distance / 100.0f;
         _last_update_ms = AP_HAL::millis();
 
         // get user configured yaw correction from front end
-        const float yaw_correction = constrain_float(frontend.get_yaw_correction(state.instance), -360.0f, +360.0f);
-        float dir_correction;
-        if (frontend.get_orientation(state.instance) == 0) {
-            dir_correction = 1.0f;
-        } else {
-            dir_correction = -1.0f;
+        const float param_yaw_offset = constrain_float(frontend.get_yaw_correction(state.instance), -360.0f, +360.0f);
```

Comment:
It's preexsisting, but why constrain this to +/- 360? `wrap_360` will manage it for you if it's outside of the range, and this can arbitrarily limit the user to what values they put in and lead to a setup problem.

Reply:
yup, good call. I'll wrap_360 instead but still constrain it to 720 just in case some crazy numbers come in

Reply:
What? No! If you need a constrain that wrap_360 has utterly failed it's role in life. You can actually not do any manipulation here and just let the wrap on the next line handle it.

Reply:
ok ok ok, no constraint at all.